src/box/sql/func.c

```diff
@@ -522,6 +522,121 @@ func_trim_str(struct sql_context *ctx, int argc, const struct Mem *argv)
 		ctx->is_aborted = true;
 }
 
+static int
+sql_utf8_symbol_count(const char *str, int byte_len)
+{
+	int symbol_count = 0;
+	int str_end = 0;
+	UChar32 c = 0;
+	while (str_end < byte_len) {
+                U8_NEXT((uint8_t *)str, str_end, byte_len, c);
+                symbol_count++;
+        }
+        return symbol_count;
+}
+
+static char*
+sql_kmp(struct sql_context *ctx, const char *str, const int str_size, 
```

Comment:
This is not one of the built-in SQL functions, so there is no point in having a context. Better to give a collation. Also, I suggest changing the return type of the function to `int` and returning `pos` if the function was successful and `-1` otherwise.

Reply:
You suggested using `uint32_t` as a return type, should I also change all similar vars inside kmp from `int` to `uint32_t`?

Reply:
Yes, I think it is good idea.